### DISCUSSION TITLE
Can I update node width by code?

### DISCUSSION THREAD

**Discussion by Eliwang4264** [xyflow/xyflow#4787] [Q&A] [answered]
Does reactFlow allow updating node width by code? When I tried to update, I got the following error<img width="298" alt="WXWorkCapture_17308701377650" src="https://github.com/user-attachments/assets/c938a877-268e-4a71-a5cf-e811002bd865">

**Comment by bcakmakoglu** [comment, marked as answer]
Yes but you shouldn't attempt to mutate node objects and instead create new ones.

Basically don't do this:
```ts
node.width = newWidth
```

Do this instead:
```ts
node = {
	...node,
	width: newWidth
}
```

**Reply by Eliwang4264** [reply]
got it  thankyou, let me try.
